src/metabase/query_processor/interface.clj

```diff
@@ -128,6 +130,11 @@
   clojure.lang.Named
   (getName [_] (name field)))
 
+(s/defrecord BinnedField [field :- Field
+                          num-bins  :- s/Int]
```

Comment:
This will need to change to include the strategy and the parameter (i.e. regex, set of bins etc). Not sure if that will just all be loaded in this BinnedField, or if that equates to different types.